app/models/concerns/inventory_computer.rb

```diff
@@ -116,6 +145,10 @@ def of(model)
         where("#{model.class.name.underscore}_id = (?)", Utils.to_id(model))
       end
 
+      def group_by(data, attr)
+        data.map { |id, value| { @model.find(id).try(attr).to_s => value } }
+      end
```

Comment:
don't think this has any place in this class, it's a util

Reply:
FYI: SumAsOf is a numeric, meaning it's like a number and supports all those operations. But it's computed on the fly when you call it.

Allowing you to do things like

```ruby
package = Package.first
package_quantity = SumAsOf.new(PackagesInventory).of(package)
puts(package_quantity) # => 10
puts(package_quantity + 1) # => 11
puts(package_quantity.as_of(1.month.ago)) #=> 5
puts(package_quantity.as_of_now) #=> 10
```

Reply:
You can look at the base class `ComputableNumeric` in `computable_numeric.rb`

Reply:
yup thanks. I moved it out of the class. I had second thoughts of keeping in that class, because the object was of StorageType, which didnt make much sense as a computation.
It Should be clear now